primitives/test-utils/src/lib.rs

```diff
@@ -67,6 +96,29 @@ pub fn make_justification_for_header<H: HeaderT>(
 	}
 }
 
+fn generate_chain<H: HeaderT>(fork_id: u8, depth: u32, ancestor: &H) -> Vec<H> {
+	let mut headers = vec![ancestor.clone()];
+
+	for i in 1..depth {
+		let parent = &headers[(i - 1) as usize];
+		let (hash, num) = (parent.hash(), *parent.number());
+
+		let mut header = test_header::<H>(num + One::one());
+		header.set_parent_hash(hash);
+
+		// Modifying the digest so headers at the same height but in different forks have different
+		// hashes
+		let digest = header.digest_mut();
+		*digest = sp_runtime::Digest {
+			logs: vec![sp_runtime::DigestItem::Other(vec![fork_id])],
```

Comment:
Can't we simply push another item instead of overwriting the entire `Digest`?
```
header.digest_mut().logs.push(DigestItem::Other(vec![fork_id]))
```
?